sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/epkversion/PartitionProcessorImpl.java

```diff
@@ -104,10 +119,17 @@ public Mono<Void> run(CancellationToken cancellationToken) {
                     }).last();
 
             })
-            .flatMap(value ->
-                        this.documentClient.createDocumentChangeFeedQuery(this.settings.getCollectionSelfLink(), this.options, itemType)
-                        .limitRequest(1)
-            )
+            .flatMap(value -> {
+                if (throughputControlGroupConfigForFeedRange != null) {
```

Comment:
This can go outside of this flatMap? Basically on the start of this method.

Reply:
This one can not, we will need to set the throughput control name for each request, because we reconstruct the options after we get the response

https://github.com/Azure/azure-sdk-for-java/blob/1c2f550b23802f10c48280b1e019c41c88a9d73f/sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/epkversion/PartitionProcessorImpl.java#L143C23-L143C23